Add edit functionality to data bags (#4942)
chef/automate: components/docs-chef-io/content/automate/chef_infra_server.md

[Data bags]({{< relref "data_bags" >}}) store global variables as JSON data. Data bags are indexed for searching and can be loaded by a cookbook or accessed during a search.

Chef Infra Server UI lets you:

- Create a data bag.
- List out all the data bags in one place.
- Search for a specific data bag item from a list of items.
- Create a data bag item.
- Edit a data bag item.
- Delete a data bag item.
- Delete a data bag.

#### Create a Data Bag

To create a new data bag, select **Create Data Bag**, as shown below:

{{< figure src="/images/automate/create-data-bag-button.png" alt="Create Data Bag Button">}}

Selecting the **Create Data Bag** button opens a dialog box. Enter the name and select **Create**.

{{< figure src="/images/automate/create-data-bag-popup.png" alt="Create Data Bag Dialog Box">}}

#### Create a Data Bag Item

> To create a data bag item, select the data bag created above and follow the steps given below:

Reviewer: Try to write each section as an independent part so users can solve a problem (in this case, "How do I create a Data Bag Item?") independently of other content.

Author: Noted! Making the possible changes to docs.

> Select **Create Item**, as shown below:

Reviewer: Remove 'as shown below' and describe the location of the button in the step or in the alt-text (don't do both. Alt-text ideally does not repeat content.

```suggestion
Select **Create Item**
```

{{< figure src="/images/automate/create-data-bag-item-button.png" alt="Create Data Bag Item Button">}}

Selecting the **Create Item** button opens a dialog box. Enter the **Data Bag Item ID** and the details of the key value pairs of that particular item. Select **Create**.

{{< figure src="/images/automate/create-data-bag-item-popup.png" alt="Create Data Bag Item Dialog Box">}}

#### Search for an Item

> Use the **Search data bag items** bar to find an item from the list of data bag items. Entering the name of an item in the search bar returns data bag items matching your search criteria.

Reviewer 2: ```suggestion
Use the **Search data bag items** bar to find an item from the list of data bag items. Entering the name of an item in the search bar returns data bag items that match your search criteria.
```

#### Edit a Data Bag Item

Select a specific data bag item to view the details. The details contain an *id*, and a couple of _key values_. Chef Infra Server lets you **edit** a data bag item.

{{< figure src="/images/automate/edit-and-delete-data-bag-item.png" alt="Edit and Delete a Data Bag Item">}}

To edit the details of the data bag items, select **Edit**. In the dialog box, you can edit the details for the specific data bag item. Once done, Select **Save Item**.

{{< figure src="/images/automate/edit-data-bag-item.png" alt="Edit a Data Bag Item">}}

#### Delete a Data Bag Item

Select a specific data bag item to view the details. The details contain an *id*, and a couple of _key values_. Chef Infra Server lets you **delete** a data bag item.

{{< figure src="/images/automate/edit-and-delete-data-bag-item.png" alt="Edit and Delete a Data Bag Item">}}

To delete a data bag item, select **Delete**. In the dialog box displayed, select **Delete** to delete the data bag item.

{{< figure src="/images/automate/delete-data-bag-item.png" alt="Delete a Data Bag Item">}}

#### Delete a Data Bag

> Chef Infra Server lets you delete the existing data bag one at a time. To delete a data bag, select the ellipses icon and then **Delete**, as illustrated below:

Reviewer 2: You might try this locally first to see how it looks, but the fontawesome shortcode will add the ellipses icon into the text. 

```suggestion
Chef Infra Server lets you delete the existing data bag one at a time. To delete a data bag, select the ellipses icon {{< fontawesome class="fas fa-ellipsis-h" >}} and then **Delete**, as illustrated below:
```

#### Search for an Item

Use the **Search data bag items** bar to find an item from the list of data bag items. Entering the name of an item in the search bar returns data bag items matching your search criteria.

{{< figure src="/images/automate/create-data-bag-item-button.png" alt="Search Data Bag Item Bar">}}

#### Edit and Delete a Data Bag Item

Select a specific data bag item to view the details. The details contain an *id*, and a couple of _key values_. Chef Infra Server lets you **edit** or **delete** a data bag item.

{{< figure src="/images/automate/edit-and-delete-data-bag-item.png" alt="Edit and Delete a Data Bag Item">}}

To edit the details of the data bag items, select **Edit**. In the dialog box, you can edit the details for the specific data bag item. Once done, Select **Save Item**.

{{< figure src="/images/automate/edit-data-bag-item.png" alt="Edit a Data Bag Item">}}

Whereas, to delete a data bag item, select **Delete**. In the dialog box displayed, select **Delete** to delete the data bag item.

{{< figure src="/images/automate/delete-data-bag-item.png" alt="Delete a Data Bag Item">}}

#### Delete a Data Bag

> Chef Infra Server lets you delete the existing data bag one at a time. To delete a data bag, select the ellipses icon and then **Delete**, as illustrated below:

Reviewer: Tautology, an ellipsis has three dots. You didn't ask for it, but here's a great [article about ellipses](https://uxdesign.cc/dot-dot-dot-7ce6170bfc7f) in UIx. Ellipsis is singular, ellipses plural. 

As to what we should call it, I think the "three dots icon" is probably clearest. @IanMadd recently upgraded us to fontawesome 5, so you have access to the icon. Ask Ian for guidance.
